resources/tenants/controller/tenants.go

```diff
@@ -34,6 +37,12 @@ func NewController(model model.Model) *Controller {
 	}
 }
 
+func NewDeploymentController(depsModel *deploymentsModel.DeploymentsModel) *Controller {
```

Comment:
There is already New function for this structure, it should handle all the init handling.
Depending on which was used to create object some methods are functional and some not is asking for trouble. When creating Controller instance all methods should be usable.

Make NewController take both models as configuration.